sdk/identity/identity/Troubleshooting.md

```diff
@@ -1,48 +1,85 @@
 
-## Troubleshoot default Azure credential authentication issues
+Errors arising from authentication issues can be thrown on any service client method that makes a request to the service. This is because the token is requested from the credential on the first call to the service and on any subsequent call that need to refresh the token.
 
-### Credential unavailable
+To distinguish these failures from failures in the service client, Azure Identity classes raise the `AuthenticationRequiredError` with details describing the source of the error in the exception message and possibly the error message. Depending on the application, these errors may or may not be recoverable.
 
-The `DefaultAzureCredential` attempts to retrieve an access token by sequentially invoking a chain of credentials. In this scenario, the `CredentialUnavailableError` signifies that all credentials in the chain failed to retrieve the token in the current environment setup/configuration. You need to follow the configuration instructions for the respective credential you're looking to use via the `DefaultAzureCredential` chain, so that the credential can work in your environment.
+```ts
+import @azure/identity
+import @azure/keyvault-secrets
+
+// Create a key client using the DefaultAzureCredential
+ const keyVaultUrl = `https://key-vault-name.vault.azure.net`;
+ const client = new KeyClient(keyVaultUrl, new DefaultAzureCredential());
+
+ try {
+// Retrieving the properties of the existing keys in that specific Key Vault.
+    console.log(await client.listPropertiesOfKeys().next());
+ } catch (error) {
+    console.log("Authentication Failed", error.message);
+ }
+```
+
+### CredentialUnavailableError
+
+The `CredentialUnavailableError` is used to indicate that the credential can’t authenticate in the current environment, due to lack of required configuration or setup. This error is also used as a signal to chained credential types, such as `DefaultAzureCredential` and `ChainedTokenCredential`, that the chained credential should continue to try other credential types later in the chain. In `ManagedIdentityCredential` it can also trigger if the authentication endpoints (like the IMDS endpoint) is unavailable.
+
+### AuthenticationError
+
+The `AuthenticationError` is used to indicate about a failure to authenticate with Azure Active Directory. The `errorResponse` field contains more details about the specific failure.
+
+## Finding Relevant Information in Error Messages
 
-Follow the configuration instructions in the "Credential unavailable error" sections of the troubleshooting guidelines below for the respective credential/authentication type you're looking to use via `DefaultAzureCredential`:
+`AuthenticationRequiredError` is thrown when unexpected errors occurred while a credential is authenticating. This can include errors received from requests to the AAD STS and often contains information helpful to diagnosis. Consider the following `AuthenticationRequiredError` message.
```

Comment:
```suggestion
`AuthenticationRequiredError` is thrown when unexpected errors occurred while a credential is authenticating. This can include errors received from requests to the AAD STS and often contains information helpful to diagnosis. Consider the following `AuthenticationRequiredError` message.
```
 AuthenticationRequiredError: invalid_request: 9002331 - [2022-02-04 00:28:06Z]: AADSTS9002331: Application '6b666991-4567-4982-9981-61877200efy1'(kaghiya-identity) is configured for use by Microsoft Account users only. Please use the /consumers endpoint to serve this request.
Trace ID: 00a7e15c-4557-4974-91d5-886428b00e00
Correlation ID: 20267531-0284-4543-93d7-cf50919fd841
Timestamp: 2022-02-04 00:28:06Z - Correlation ID: 20267531-0284-4543-93d7-cf50919fd841 - Trace ID: 00a7e15c-4557-4974-91d5-886428b00e00
```
```